# Analisis de datos sobre la venta de videojuegos

![](https://static1.howtogeekimages.com/wordpress/wp-content/uploads/2024/04/shutterstock_1290635251.jpg?q=49&fit=crop&w=1100&h=618&dpr=2)

## Introducción

Este conjunto de datos ofrece información sobre las ventas de videojuegos según plataforma, editora y género para los títulos más populares a nivel mundial.
Con más de 16,000 registros de ventas entre 1980 y 2016, este análisis busca descubrir relaciones clave entre estas variables y proporcionar insights sobre las tendencias de una industria que mueve millones de dólares cada año.

## Objetivo

Este estudio tiene como objetivo analizar cuáles plataformas lideran las ventas globales y por región, identificar los géneros que predominan en las preferencias de los jugadores, y observar cómo estos factores han cambiado a lo largo del tiempo.
Este análisis ayudará a comprender la evolución de la industria de los videojuegos y a identificar tendencias clave que podrían ayudar a los desarrolladores y editores a alcanzar el éxito en el futuro.

## Hipótesis

1. ¿Cuáles son los juegos, géneros, plataformas y editoriales más populares en términos de cantidad de publicaciones?
2. Correlación de datos
3. ¿Cuáles son las cifras de ventas por región y a nivel mundial de los videojuegos?
4. ¿Cómo se distribuye el consumo de videojuegos a nivel mundial? ¿Qué regiones representan el mayor porcentaje de las ventas globales y por qué?
5. ¿Cuáles son los videojuegos más vendidos por región y a nivel mundial?
6. ¿Cuáles son las plataformas de videojuegos más populares en términos de ventas por región y a nivel mundial?
7. ¿Cuáles son los géneros de videojuegos más vendidos por región y a nivel mundial?
8. ¿Cuáles son las editoriales de videojuegos con mayor volumen de ventas por región y a nivel mundial?
9. ¿Cómo ha evolucionado la popularidad de los diferentes géneros de videojuegos a lo largo de los años?
10. ¿Existe alguna correlación entre las preferencias de los jugadores en diferentes regiones y los factores demográficos, culturales o económicos?
11. ¿Cómo se comparan las ventas de videojuegos por género, plataforma y editorial en diferentes regiones?

|Feature|Descripción |tipo de dato |unidade de medida|clasificación de dato |tratamiento |
|---|--- |---- |--- |--- |--- |
|``Rank``|  Ranking del juego basado en las ventas globales. |int | |indice | |
|``Name``| Nombre del juego. |obj | |cualitativo nominal |Tratar una fila con duplicado |
|``Platform``| Plataforma en la que se lanzó el juego. |obj | |cualitativo nominal | |
|``Year``| Año de lanzamiento del juego. |int | |cuantitativo discreto |Tratar missings, rellenar los dados faltantes |
|``Genre``| Género del juego. |obj | |cualitativo nominal | |
|``Publisher``| Editor del juego. |obj | |cualitativo nominal |Tratar missings, rellenar los dados faltantes |
|``NA_Sales``|Ventas del juego en Norteamérica. |float |millones |cuantitativo continuo | |
|``EU_Sales``|Ventas del juego en Europa.|float |millones |cuantitativo continuo | |
|``JP_Sales``|Ventas del juego en Japón. |float |millones |cuantitativo continuo | |
|``Other_Sales``|Ventas del juego en otras regiones. |float |millones |cuantitativo continuo | |
|``Global_Sales``| Ventas totales del juego en todo el mundo. |float |millones |cuantitativo continuo | |

# Preparación y carga de datos

## Librerias

In [41]:
import os
import sys

root_path = os.path.abspath(r'C:\Users\Felipe\Desktop\THE-BRIDGE\THEBRIDGE-BOOTCAMP\PROYECTO_EDA_VGSALES\proyecto_eda_ventas_videojuegos')
sys.path.append(root_path)
vgsales_path = os.path.join(root_path, 'data', 'raw', 'vgsales.csv')
scripts_path = os.path.join(root_path, 'data', 'scripts', 'auto_importer.py')


In [42]:
print(sys.path)

['C:\\Program Files\\Python311\\python311.zip', 'C:\\Program Files\\Python311\\DLLs', 'C:\\Program Files\\Python311\\Lib', 'C:\\Program Files\\Python311', 'c:\\Users\\Felipe\\AppData\\Local\\pypoetry\\Cache\\virtualenvs\\proyecto-eda-ventas-videojuegos-znrmrGGq-py3.11', '', 'c:\\Users\\Felipe\\AppData\\Local\\pypoetry\\Cache\\virtualenvs\\proyecto-eda-ventas-videojuegos-znrmrGGq-py3.11\\Lib\\site-packages', 'c:\\Users\\Felipe\\AppData\\Local\\pypoetry\\Cache\\virtualenvs\\proyecto-eda-ventas-videojuegos-znrmrGGq-py3.11\\Lib\\site-packages\\win32', 'c:\\Users\\Felipe\\AppData\\Local\\pypoetry\\Cache\\virtualenvs\\proyecto-eda-ventas-videojuegos-znrmrGGq-py3.11\\Lib\\site-packages\\win32\\lib', 'c:\\Users\\Felipe\\AppData\\Local\\pypoetry\\Cache\\virtualenvs\\proyecto-eda-ventas-videojuegos-znrmrGGq-py3.11\\Lib\\site-packages\\Pythonwin', 'C:\\Users\\Felipe\\Desktop\\THE-BRIDGE\\THEBRIDGE-BOOTCAMP\\PROYECTO_EDA_VGSALES\\proyecto_eda_ventas_videojuegos', 'C:\\Users\\Felipe\\Desktop\\THE-B

In [43]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from pathlib import Path
from scripts.auto_importer import AutoImporter
from scripts.auto_importer import DataFrameDescriber

## Carga de datos

In [44]:
vgsales = pd.read_csv(vgsales_path, index_col=0)

In [45]:
df = vgsales.copy()

In [46]:
auto_importer = AutoImporter(df)
df_descripcion = DataFrameDescriber(df)

# Primera exploración

In [47]:
df.head()

Name Platform    Year         Genre Publisher  \
Rank                                                                      
1                   Wii Sports      Wii  2006.0        Sports  Nintendo   
2            Super Mario Bros.      NES  1985.0      Platform  Nintendo   
3               Mario Kart Wii      Wii  2008.0        Racing  Nintendo   
4            Wii Sports Resort      Wii  2009.0        Sports  Nintendo   
5     Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing  Nintendo   

      NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
Rank                                                           
1        41.49     29.02      3.77         8.46         82.74  
2        29.08      3.58      6.81         0.77         40.24  
3        15.85     12.88      3.79         3.31         35.82  
4        15.75     11.01      3.28         2.96         33.00  
5        11.27      8.89     10.22         1.00         31.37

In [48]:
df.tail()

Name Platform    Year  \
Rank                                                                       
16596                Woody Woodpecker in Crazy Castle 5      GBA  2002.0   
16597                     Men in Black II: Alien Escape       GC  2003.0   
16598  SCORE International Baja 1000: The Official Game      PS2  2008.0   
16599                                        Know How 2       DS  2010.0   
16600                                  Spirits & Spells      GBA  2003.0   

          Genre   Publisher  NA_Sales  EU_Sales  JP_Sales  Other_Sales  \
Rank                                                                     
16596  Platform       Kemco      0.01      0.00       0.0          0.0   
16597   Shooter  Infogrames      0.01      0.00       0.0          0.0   
16598    Racing  Activision      0.00      0.00       0.0          0.0   
16599    Puzzle    7G//AMES      0.00      0.01       0.0          0.0   
16600  Platform     Wanadoo      0.01      0.00       0.0          0.0   

       Global_Sales  
Rank                 
16596          0.01  
16597          0.01  
16598          0.01  
16599          0.01  
16600          0.01

In [49]:
df.shape

(16598, 10)

In [50]:
auto_importer.inspeccion_inicial()

=== TAMAÑO Y ESTRUCTURA DE LOS DATOS ===
Número total de registros: 16598
Número de columnas: 10
Uso de memoria: 1426.39 KB


=== TIPOS DE DATOS Y NOMBRES DE COLUMNAS ===
Name             object
Platform         object
Year            float64
Genre            object
Publisher        object
NA_Sales        float64
EU_Sales        float64
JP_Sales        float64
Other_Sales     float64
Global_Sales    float64
dtype: object


Información detallada del DataFrame:
<class 'pandas.core.frame.DataFrame'>
Index: 16598 entries, 1 to 16600
Data columns (total 10 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Name          16598 non-null  object 
 1   Platform      16598 non-null  object 
 2   Year          16327 non-null  float64
 3   Genre         16598 non-null  object 
 4   Publisher     16540 non-null  object 
 5   NA_Sales      16598 non-null  float64
 6   EU_Sales      16598 non-null  float64
 7   JP_Sales      16598 non-null  float64
 8

In [51]:
df.isna().sum()

Name              0
Platform          0
Year            271
Genre             0
Publisher        58
NA_Sales          0
EU_Sales          0
JP_Sales          0
Other_Sales       0
Global_Sales      0
dtype: int64

In [52]:
df.shape[0]

16598

In [53]:
58 / df.shape[0] * 100

0.3494396915290999

## Limpieza y preprocesado

In [54]:
# Objeto con los datos para llenar la columna Publisher donde tenía dados igual a 'Unknown'

gamepublisher_unknown = {
    942: "Sony Computer Entertainment",
    1651: "EA Sports",
    2116: "Konami",
    2175: "Hasbro Interactive",
    2238: "Konami",
    2288: "Namco",
    2297: "Nintendo",
    2330: "Midway Games",
    2486: "EA Sports",
    2493: "Hasbro Interactive",
    2499: "Midway Games",
    2546: "Square Enix",
    2778: "EA Sports",
    2788: "Nintendo",
    2831: "Bandai Namco Games",
    3032: "Trion Worlds",
    3470: "Infogrames",
    3503: "EA Sports",
    3704: "TopWare Interactive",
    4123: "TopWare Interactive",
    4538: "EA Sports",
    4685: "En Masse Entertainment",
    4799: "989 Sports",
    4867: "Natsume",
    5173: "PopCap Games",
    5197: "3DO",
    5620: "Square Enix",
    5627: "Activision",
    5671: "Acclaim Entertainment",
    5863: "Ubisoft",
    5915: "Natsume",
    6023: "Taito",
    6044: "THQ",
    6213: "Namco",
    6278: "DSI Games",
    6403: "Konami",
    6409: "Nintendo",
    6649: "2K Sports",
    6779: "Capcom",
    6834: "Activision",
    6921: "NIS America",
    6970: "Sony Computer Entertainment",
    7037: "Sierra Entertainment",
    7255: "Yoostar Entertainment",
    7316: "Natsume",
    7374: "Nordcurrent",
    7384: "TDK Mediactive",
    7523: "TopWare Interactive",
    7938: "Nintendo",
    8046: "Atlus",
    8088: "505 Games",
    8138: "Mastiff",
    8159: "Sony Computer Entertainment",
    8169: "Konami",
    8206: "Sega",
    8225: "THQ",
    8315: "505 Games",
    8507: "D3 Publisher",
    8622: "Data Design Interactive",
    8709: "Telltale Games",
    8827: "Mentor Interactive",
    8846: "505 Games",
    8931: "Acclaim Entertainment",
    8977: "Nippon Ichi Software",
    9083: "Majesco Entertainment",
    9153: "Electronic Arts",
    9173: "PopCap Games",
    9204: "Mattel Interactive",
    9207: "Seven45 Studios",
    9253: "PopCap Games",
    9524: "Nintendo",
    9669: "Electronic Arts",
    9771: "Sony Computer Entertainment",
    9817: "Activision",
    9822: "Majesco Entertainment",
    9823: "Oxygen Games",
    9842: "Ubisoft",
    9870: "2K Sports",
    9884: "Seven45 Studios",
    9999: "PopCap Games",
    10196: "Yoostar Entertainment",
    10244: "Kadokawa Shoten",
    10274: "Nintendo",
    10359: "Imageepoch",
    10556: "Nintendo",
    10561: "Ubisoft",
    10735: "Atlus",
    10760: "Marvelous Entertainment",
    10794: "Oxygen Games",
    11109: "THQ",
    11277: "Mastiff",
    11325: "Electronic Arts",
    11457: "1st Playable Productions",
    11545: "Konami",
    11602: "HB Studios",
    11800: "Namco",
    11891: "Data Design Interactive",
    12017: "Majesco Entertainment",
    12029: "Majesco Entertainment",
    12061: "Konami",
    12129: "The Game Factory",
    12137: "Yoostar Entertainment",
    12186: "Ubisoft",
    12214: "Black Bean Games",
    12300: "Press Play",
    12309: "Nintendo",
    12533: "Destineer",
    12535: "Mastiff",
    12668: "Majesco Entertainment",
    12751: "Ubisoft",
    12881: "The Learning Company",
    12900: "MumboJumbo",
    12917: "Valcon Games",
    12931: "Nintendo",
    13071: "Mentor Interactive",
    13113: "Kadokawa Shoten",
    13145: "MumboJumbo",
    13473: "Aksys Games",
    13587: "Press Play",
    13644: "iWin",
    13711: "1C Company",
    13735: "Midway Games",
    13738: "Destineer",
    13759: "Evolved Games",
    13812: "Storm City Games",
    13822: "Level-5",
    13872: "Sega",
    13932: "Dovetail Games",
    13985: "Team17",
    14032: "G5 Entertainment",
    14048: "Mastiff",
    14058: "Gaslamp Games",
    14100: "2K Sports",
    14111: "Fields",
    14158: "Teyon",
    14183: "Avanquest Software",
    14252: "Nicalis",
    14257: "THQ",
    14276: "Konami",
    14296: "Konami",
    14315: "Paradox Interactive",
    14355: "Warner Bros. Interactive Entertainment",
    14374: "Headup Games",
    14380: "Namco Bandai Games",
    14407: "Bohemia Interactive",
    14476: "Destineer",
    14525: "SNK Playmore",
    14564: "Fields",
    14579: "Alawar Entertainment",
    14620: "Fields",
    14643: "Sprite",
    14679: "NewKidCo",
    14699: "Midway Games",
    14763: "HB Studios",
    14777: "Easy Interactive",
    14847: "Paradox Interactive",
    14848: "PopCap Games",
    14852: "Atlus",
    14858: "SNK Playmore",
    14928: "NEC Interchannel",
    14937: "Oxygen Games",
    15001: "Encore Software",
    15066: "Hudson Soft",
    15160: "Spike Chunsoft",
    15166: "Paradox Interactive",
    15180: "Vridge",
    15187: "Plum",
    15200: "Deep Silver",
    15211: "Atlus",
    15270: "Trendy Entertainment",
    15276: "THQ",
    15326: "Experience Inc.",
    15338: "Astragon",
    15347: "Rondomedia",
    15351: "Konami",
    15454: "UFO Interactive Games",
    15479: "Konami",
    15551: "Mastiff",
    15567: "Bushiroad",
    15582: "Activision",
    15609: "IE Institute",
    15655: "Sega",
    15741: "Hasbro Interactive",
    15839: "Big Fish Games",
    15868: "Oxygen Games",
    15879: "TDK Mediactive",
    16005: "GungHo Online Entertainment",
    16068: "TDK Mediactive",
    16149: "Nippon Ichi Software",
    16174: "Big Fish Games",
    16197: "Warner Bros. Interactive Entertainment",
    16203: "Idea Factory",
    16238: "D3 Publisher",
    16289: "Brightrock Games",
    16310: "Midway Games",
    16369: "Idea Factory",
    16427: "Virtual Playground",
    16430: "Sega",
    16463: "Ravenscourt",
    16496: "Ubisoft",
    16535: "Paradox Interactive",
    16548: "Alawar Entertainment",
    16584: "Deep Silver"
}

In [55]:
# Objeto con datos para insertar en la columna publisher que tenía valores nulos

game_publisher = {
    471: "THQ",
    1305: "EA Sports",
    1664: "Majesco Entertainment",
    2224: "Sony Computer Entertainment",
    3161: "Majesco Entertainment",
    3168: "Majesco Entertainment",
    3768: "Majesco Entertainment",
    4147: "Sega",
    4528: "Majesco Entertainment",
    4637: "Majesco Entertainment",
    5304: "Bandai Namco Games",
    5649: "Majesco Entertainment",
    6274: "Nintendo",
    6439: "Majesco Entertainment",
    6564: "Majesco Entertainment",
    6650: "Majesco Entertainment",
    6851: "Majesco Entertainment",
    7210: "Majesco Entertainment",
    7353: "Atari",
    7472: "Majesco Entertainment",
    7955: "Majesco Entertainment",
    8332: "Majesco Entertainment",
    8343: "Majesco Entertainment",
    8370: "Majesco Entertainment",
    8505: "SouthPeak Games",
    8772: "Majesco Entertainment",
    8850: "Majesco Entertainment",
    8898: "Majesco Entertainment",
    9519: "Astragon",
    9751: "Bandai Namco Games",
    10384: "Majesco Entertainment",
    10496: "Konami",
    11078: "Hasbro Interactive",
    11528: "Majesco Entertainment",
    12489: "Nintendo",
    12519: "Nippon Ichi Software",
    13280: "Capcom",
    13674: "Hudson Soft",
    13964: "Wargaming.net",
    14089: "Nintendo",
    14299: "Hudson Soft",
    14314: "GameMill Entertainment",
    14701: "Excalibur Publishing",
    14945: "Alchemist",
    15059: "Activision",
    15264: "Nintendo",
    15328: "Majesco Entertainment",
    15356: "Astragon",
    15791: "Idea Factory",
    15918: "Aspyr",
    16194: "Gearbox Software",
    16201: "Bandai Namco Games",
    16211: "Majesco Entertainment",
    16370: "Rejet",
    16497: "Majesco Entertainment",
    16546: "Astragon",
    16556: "Focus Home Interactive"
}

In [56]:
# Objeto que contiene datos que se insertarán en la columna año en líneas donde el valor era incorrecto o nulo

game_years = {
    16201: 2011,
    11078: 2000,
    3219: 2008,
    1515: 1980,
    16249: 2008,
    2115: 1977,
    12131: 2006,
    5671: 2004,
    8931: 2004,
    8555: 2001,
    16068: 2003,
    10496: 2007,
    13674: 1998,
    8315: 2010,
    8088: 2010,
    13057: 2011,
    14586: 2012,
    9253: 2010,
    5173: 2010,
    9173: 2010,
    14276: 2012,
    6968: 2010,
    14299: 2007,
    6808: 2010,
    7372: 1983,
    16232: "Canceled",
    12129: 2010,
    15582: 2006,
    713: 2006,
    1128: 2010,
    12827: 2011,
    12628: 2011,
    9669: 2009,
    11325: 2009,
    12489: 2006,
    4234: 1980,
    14058: 2016,
    6649: 2005,
    1587: 1977,
    12533: 2005,
    14476: 2005,
    14879: 2012,
    5197: 2001,
    5510: 1999,
    15881: 2010,
    6928: 2011,
    6921: 2010,
    11575: 2013,
    7650: 2013,
    3717: 2004,
    10999: 2005,
    15879: 2003,
    5872: 2012,
    10364: 2006,
    9312: 2005,
    2021: 1997,
    5304: 2006,
    5659: 1980,
    12017: 2003,
    15918: 2008,
    11923: 2011,
    12931: 2005,
    10831: 2005,
    378: 2003,
    11378: 2011,
    7617: 1997,
    15814: 2010,
    5620: 2002,
    3882: 1980,
    6318: 1978,
    12668: 2010,
    15742: 2006,
    16310: 2003,
    650: 2001,  
    14380: 2010,
    12809: 2008,
    13667: 2008,
    11333: 2010,
    7371: 1992, 
    10274: 2003,
    9822: 2002, 
    6316: 2005, 
    16369: 2010,
    14057: 2011,
    5063: 1980, 
    9983: 2011, 
    11144: 2011,
    10467: 2011,
    10692: 2011,
    4867: 2001, 
    5915: 2011, 
    6127: 2002, 
    1699: 2002, 
    8762: 1978, 
    16194: 2015,
    14089: 2006,
    6285: 1977, 
    16330: 2012,
    8336: 2002, 
    15903: 2007,
    6970: 2001, 
    12250: 2011,
    6498: 2011, 
    8709: 2011, 
    4153: 1982,
    432: 2008,   # LEGO Batman: The Videogame (Wii)
    1500: 2008,  # LEGO Batman: The Videogame (PSP)
    7215: 2011,  # LEGO Harry Potter: Years 5-7 (PC)
    2147: 2011,  # LEGO Harry Potter: Years 5-7 (PS3)
    2088: 2011,  # LEGO Harry Potter: Years 5-7 (X360)
    1435: 2011,  # LEGO Harry Potter: Years 5-7 (Wii)
    4793: 2011,  # LEGO Harry Potter: Years 5-7 (3DS)
    6321: 2011,  # LEGO Harry Potter: Years 5-7 (PSP)
    2840: 2011,  # LEGO Harry Potter: Years 5-7 (DS)
    653: 2008,   # LEGO Indiana Jones: The Original Adventures (Wii)
    1992: 1999,  # Legacy of Kain: Soul Reaver (PS)
    14852: 2009, # Legacy of Ys: Books I & II (DS)
    10760: 2008, # Luminous Arc 2 (JP sales) (DS)
    13677: 2008, # Luxor: Pharaoh's Challenge (Wii)
    2499: 2002,  # MLB SlugFest 20-03 (PS2)
    3423: 2005,  # Madden NFL 06 (X360)
    2486: 2006,  # Madden NFL 07 (PSP)
    2778: 2010,  # Madden NFL 11 (Wii)
    3503: 2001,  # Madden NFL 2002 (XB)
    180: 2003,   # Madden NFL 2004 (PS2)
    14315: 2009, # Majesty 2: The Fantasy Kingdom Sim (X360)
    9870: 2006,  # Major League Baseball 2K6 (PSP)
    14100: 2008, # Major League Baseball 2K8 (PSP)
    15264: 2012, # Mario Tennis (3DS)
    4380: 1980,  # Maze Craze: A Game of Cops 'n Robbers (2600)
    11978: 2004, # McFarlane's Evil Prophecy (PS2)
    11757: 2009, # Mega Man Battle Network: Operation Shooting Star (DS)
    6779: 2006,  # Mega Man X Collection (PS2)
    3240: 2002,  # Metal Gear Solid 2: Substance (XB)
    12711: 2008, # Mobile Ops: The One Year War (X360)
    3196: 2007,  # Monster Hunter 2 (PS2)
    13280: 2013, # Monster Hunter Frontier Online (PS3)
    12917: 2007, # Mountain Bike Adrenaline (PS2)
    7183: 2011,  # Move Fitness (PS3)
    10561: 2009,  # My Healthy Cooking Coach (DS)  
    1651: 2002,
    3470: 2002,
    5903: 2002,
    5164: 2002,
    8206: 2002,
    3755: 2003,
    4799: 2002,
    13735: 2003,
    4538: 2010,
    2288: 2002,
    12214: 2011,
    6044: 2006,
    9524: 2003,
    8169: 2011,
    9083: 2010,
    16060: 2007,
    15868: 2007,
    2588: 2008,
    6213: 2002,
    15001: 2005,
    7910: 2011,
    12240: 2012,
    12030: 2012,
    12519: 2009,
    12186: 2002,
    12881: 2010,
    9744: 2011,
    2297: 2008,
    4474: 2008,
    4936: 2008,
    7384: 2002,
    1135: 2007,
    784: 2007,
    625: 2007,
    1839: 2007,
    6403: 2008, # Rock Revolution PS3
    11545: 2008, # Rock Revolution Wii
    9842: 2008, # Rocksmith PC
    13479: 2009, # RollerCoaster Tycoon PC
    15319: 2009, # Runaway: A Twist of Fate DS
    14162: 2004, # Sabre Wulf GBA
    15200: 2007, # Saint Wii
    10981: 2008, # Samurai Shodown Anthology PSP
    14525: 2007, # Samurai Spirits: Tenkaichi Kenkakuden PS2
    9771: 2002, # Saru! Get You! Million Monkeys PS2
    13872: 2006, # Sega Rally 2006 PS2
    2530: 2008, # Shaun White Snowboarding X360
    15700: 2007, # Shonen Jump's Yu-Gi-Oh! GX Card Almanac DS
    16197: 2008, # Shorts DS
    5840: 2007, # Shrek the Third DS
    4860: 2008, # Silent Hill: Homecoming PS3
    4382: 2008, # Silent Hill: Homecoming X360
    6834: 2010, # Singularity PS3
    5627: 2010, # Singularity X360
    5800: 1978, # Slot Machine 2600
    11800: 2001, # Smashing Drive GC
    4147: 2006, # Sonic the Hedgehog PS3
    608: 1978, # Space Invaders 2600
    8441: 2010, # Splatterhouse PS3
    10047: 2007, # Star Trek: Conquest PS2
    7948: 2006, # Star Trek: Legacy X360
    5771: 2003, # Star Wars Jedi Knight II: Jedi Outcast XB
    15211: 2009, # Steal Princess DS
    9477: 2009, # Street Fighter IV PC
    9817: 2002, # Street Hoops GC
    2116: 2002, # Suikoden III PS2
    4471: 1978, # Super Breakout 2600
    14699: 2001, # Super Duper Sumos GBA
    11110: 2001, # Super Puzzle Fighter II GBA
    9751: 2009,
    8378: 2002,
    14048: 2009,
    4685: 2011,
    8225: 2006,
    3143: 2011,
    7722: 2011,
    2949: 2011,
    6197: 2005,
    3051: 2005,
    4332: 2004,
    9823: 2007,
    10794: 2007,
    5863: 2004,
    5080: 2007,
    3348: 2007,
    11457: 2011,
    13158: 2009,
    14858: 2004,
    6274: 2004,
    2424: 2011,
    2417: 2011,
    16496: 2011,
    4201: 2006,
    12751: 2006,
    14679: 2002,
    5242: 2013,
    6939: 2006,
    11940: 2010,
    12924: 2011,
    9394: 2002,
    8046: 2010,
    7037: 2002,
    1305: 1998,
    5198: 2011,
    14928: 2003,
    8159: 2001,
    14945: 2011,
    6155: 2004,
    7408: 2010,
    8622: 2006,
    15655: 2003,
    16430: 2003,
    9153: 2000,
    7425: 2010,
    9694: 2010,
    16061: 2010,
    2788: 2004,
    3954: 2010,
    2175: 2002,
    15000: 2006,
    15002: 2006,
    15720: 2005,
    13964: 2011,
    15609: 2007,
    2136: 2010,
    12137: 2008,
    7353: 2002,
    15479: 2009,
    11411: 2007,
    8901: 2002,
    471: 2005,
    15870: 2010
}

iteración para arreglar la columna ``Year``

In [57]:
for n in df.index:
    if n in game_years.keys():
        df.at[n, 'Year'] = game_years[n]

C:\Users\Felipe\AppData\Local\Temp\ipykernel_27000\1049631621.py:3: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value 'Canceled' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.
  df.at[n, 'Year'] = game_years[n]


iteraciones para arreglar la columna ``Publisher``

In [58]:
for n in df.index:
    if n in game_publisher.keys():
        df.at[n, 'Publisher'] = game_publisher[n]

In [59]:
for n in df.index:
    if n in gamepublisher_unknown.keys():
        df.at[n, 'Publisher'] = gamepublisher_unknown[n]

Comprobando si queda algún valor nulo

In [60]:
df.isna().sum()

Name            0
Platform        0
Year            0
Genre           0
Publisher       1
NA_Sales        0
EU_Sales        0
JP_Sales        0
Other_Sales     0
Global_Sales    0
dtype: int64

Eliminar el dato nulo restante debido a que es un juego que no ha sido lanzado.

In [61]:
df = df[df['Year'] != "Canceled"]

Confermando la inexistencia de nulos.

In [62]:
df.isna().sum()

Name            0
Platform        0
Year            0
Genre           0
Publisher       0
NA_Sales        0
EU_Sales        0
JP_Sales        0
Other_Sales     0
Global_Sales    0
dtype: int64

Durante la inserción de datos de la columna ``Year`` visando quitar los valores nulos de esta columna, el juego que no ha sido lanzado, contenía un texto en lugar de un número entero, cambiando el tipo de datos de toda la columna, por lo tanto tengo que cambiar el tipo de datos otra vez a numericos utilizando el siguiente codigo:

In [63]:
df['Year'] = pd.to_numeric(df['Year'])
df['Year'] = df['Year'].astype(int)

df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 16597 entries, 1 to 16600
Data columns (total 10 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Name          16597 non-null  object 
 1   Platform      16597 non-null  object 
 2   Year          16597 non-null  int64  
 3   Genre         16597 non-null  object 
 4   Publisher     16597 non-null  object 
 5   NA_Sales      16597 non-null  float64
 6   EU_Sales      16597 non-null  float64
 7   JP_Sales      16597 non-null  float64
 8   Other_Sales   16597 non-null  float64
 9   Global_Sales  16597 non-null  float64
dtypes: float64(5), int64(1), object(4)
memory usage: 1.4+ MB


C:\Users\Felipe\AppData\Local\Temp\ipykernel_27000\2517050206.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['Year'] = pd.to_numeric(df['Year'])
C:\Users\Felipe\AppData\Local\Temp\ipykernel_27000\2517050206.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['Year'] = df['Year'].astype(int)


Comprobando se hay valores duplicados en el dataset

In [64]:
df[df.duplicated(keep=False)]

Name Platform  Year    Genre Publisher  \
Rank                                                                   
15000  Wii de Asobu: Metroid Prime      Wii  2006  Shooter  Nintendo   
15002  Wii de Asobu: Metroid Prime      Wii  2006  Shooter  Nintendo   

       NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
Rank                                                            
15000       0.0       0.0      0.02          0.0          0.02  
15002       0.0       0.0      0.02          0.0          0.02

Eliminando valor duplicado

In [65]:
df = df.drop_duplicates()

Comprobando que la línea duplicada ya no está presente en el dataset

In [66]:
df.query('Name.str.contains("Wii de Asobu: Metroid Prime")')

Name Platform  Year    Genre Publisher  \
Rank                                                                   
15000  Wii de Asobu: Metroid Prime      Wii  2006  Shooter  Nintendo   

       NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
Rank                                                            
15000       0.0       0.0      0.02          0.0          0.02

Aquí se hace un filtro para eliminar los años con reducida muestra de datos, mantener estes datos podría causar distorción de los resultados, no presentando correctamente las tendencias del mercado para dichos años y llevándonos a conclusiones equivocadas.

In [67]:
df.sort_values(by='Year', ascending=False).tail(10)

Name Platform  Year     Genre   Publisher  NA_Sales  EU_Sales  \
Rank                                                                            
2671          Boxing     2600  1980  Fighting  Activision      0.72      0.04   
259        Asteroids     2600  1980   Shooter       Atari      4.00      0.26   
8762        Home Run     2600  1978    Sports       Atari      0.14      0.01   
6318    Flag Capture     2600  1978    Action       Atari      0.25      0.02   
4471  Super Breakout     2600  1978    Puzzle       Atari      0.41      0.03   
608   Space Invaders     2600  1978   Shooter       Atari      2.36      0.14   
5800    Slot Machine     2600  1978    Action       Atari      0.29      0.02   
6285        Indy 500     2600  1977    Racing       Atari      0.26      0.01   
1587          Combat     2600  1977    Action       Atari      1.17      0.07   
2115  Air-Sea Battle     2600  1977   Shooter       Atari      0.91      0.06   

      JP_Sales  Other_Sales  Global_Sales  
Rank                                       
2671       0.0         0.01          0.77  
259        0.0         0.05          4.31  
8762       0.0         0.00          0.15  
6318       0.0         0.00          0.27  
4471       0.0         0.00          0.44  
608        0.0         0.03          2.53  
5800       0.0         0.00          0.31  
6285       0.0         0.00          0.27  
1587       0.0         0.01          1.25  
2115       0.0         0.01          0.98

In [68]:
df.sort_values(by='Year', ascending=False).head(10)

Name Platform  Year  \
Rank                                                                      
5959                              Imagine: Makeup Artist       DS  2020   
16441                   Brothers Conflict: Precious Baby      PSV  2017   
14393   Phantasy Star Online 2 Episode 4: Deluxe Package      PS4  2017   
16244   Phantasy Star Online 2 Episode 4: Deluxe Package      PSV  2017   
16579                                  Rugby Challenge 3     XOne  2016   
16592  Chou Ezaru wa Akai Hana: Koi wa Tsuki ni Shiru...      PSV  2016   
12901                                     Dark Souls III       PC  2016   
16558              Codename: Panzers Complete Collection       PC  2016   
16553  God Eater Off Shot: Tachibana Sakuya-hen Twin ...      PS4  2016   
16533                           Sébastien Loeb Rally Evo     XOne  2016   

              Genre             Publisher  NA_Sales  EU_Sales  JP_Sales  \
Rank                                                                      
5959     Simulation               Ubisoft      0.27      0.00      0.00   
16441        Action          Idea Factory      0.00      0.00      0.01   
14393  Role-Playing                  Sega      0.00      0.00      0.03   
16244  Role-Playing                  Sega      0.00      0.00      0.01   
16579        Sports  Alternative Software      0.00      0.01      0.00   
16592        Action       dramatic create      0.00      0.00      0.01   
12901  Role-Playing    Namco Bandai Games      0.00      0.05      0.00   
16558      Strategy          Nordic Games      0.00      0.01      0.00   
16553        Action    Namco Bandai Games      0.00      0.00      0.01   
16533        Racing       Milestone S.r.l      0.00      0.01      0.00   

       Other_Sales  Global_Sales  
Rank                              
5959          0.02          0.29  
16441         0.00          0.01  
14393         0.00          0.03  
16244         0.00          0.01  
16579         0.00          0.01  
16592         0.00          0.01  
12901         0.00          0.05  
16558         0.00          0.01  
16553         0.00          0.01  
16533         0.00          0.01

In [69]:
df = df.drop(df[(df['Year'] == 1977) | (df['Year'] == 1978) | (df['Year'] == 2017) | (df['Year'] == 2020)].index)

In [70]:
df.query('Name.str.contains("Bentley")')

Name Platform  Year Genre                    Publisher  \
Rank                                                                         
2224  Bentley's Hackpack      GBA  2005  Misc  Sony Computer Entertainment   

      NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
Rank                                                           
2224      0.67      0.25       0.0         0.02          0.93

Para evitar problemas eliminaremos este registro debido a que los datos de la columna ``Platform`` no son correctos.

In [71]:
df = df.drop(df[df['Name'] == "Bentley's Hackpack"].index)

In [72]:
df.shape

(16583, 10)

In [81]:
df.to_csv(r'..\data\processed\vgsales_limpio.csv')

# Conclusion

Se puede observar que el dataframe está composto de 16583 filas y 10 columnas, la columna rank se convertió en índice, dado que es una columna que no se repite y designa la clasificación de cada juego. Los missings fueron tratados, a través de insvestigación fue posible llenar la base de datos de manera satisfactoria, evitando eliminar datos o insertar datos con valores predichos desde los datos presentes, también se eliminó del conjunto de datos un juego cuyo lanzamiento fue cancelado, un juego cuyos  datos no coincidían con la información correcta, un registro duplicado y los registros de los años 1980, 1978, 2017 y 2020 debido a la baja cuantidad de datos, para que se pueda trabajar en la serie temporal completa en lugar de usar datos com una muestra reducida que es un factor que podría no dismostrar con precisión las tendencias de mercado en el determinado período.

# Extra

Creación de un dataframe agrupado por ``genero`` y ``año`` para possibilitar la creación de un bar chart race para muestrar los generos de juegos más vendidos al largo de los años.

In [82]:
df_grouped = df.groupby(['Year', 'Genre'], as_index=False)['Global_Sales'].sum()

df_reshaped = pd.melt(df_grouped, id_vars=['Year', 'Genre'], value_vars=['Global_Sales'], 
                      var_name='Sales_Type', value_name='Value')

In [83]:
df_pivoted = df_reshaped.pivot_table(index='Genre', columns='Year', values='Value', aggfunc='sum')

df_pivoted = df_pivoted.reset_index()

In [84]:
df_cumulative = df_pivoted.set_index('Genre').cumsum(axis=1)

In [85]:
df_cumulative.to_csv(r'..\data\processed\cumulative_sales_bar_chart_race.csv')

Creación de un dataframe para una presentación de un treemap chart de los top 10 de ventas por editora.

In [86]:
top_10_global_sales_publisher = df.groupby('Publisher')[['Global_Sales']].sum().sort_values(by='Global_Sales',ascending=False).reset_index()

In [87]:
top_10_global_sales_publisher = top_10_global_sales_publisher.head(10)
top_10_global_sales_publisher

Publisher  Global_Sales
0                      Nintendo       1789.48
1               Electronic Arts       1110.66
2                    Activision        728.18
3   Sony Computer Entertainment        609.87
4                       Ubisoft        475.09
5          Take-Two Interactive        399.54
6                           THQ        344.37
7  Konami Digital Entertainment        283.64
8                          Sega        273.68
9            Namco Bandai Games        254.12

In [88]:
top_10_global_sales_publisher.to_csv(r'..\data\processed\top10publisherglobal.csv')